src/sentry/incidents/serializers/alert_rule.py

```diff
@@ -313,6 +313,11 @@ def update(self, instance, validated_data):
                 )
                 raise BadRequest
             self._handle_triggers(alert_rule, triggers)
+            try:
+                dual_update_alert_rule(alert_rule)
+            except Exception:
+                sentry_sdk.capture_exception()
+                raise BadRequest(message="Error when updating alert rule")
```

Comment:
```suggestion
            if should_dual_write:
                try:
                    dual_update_alert_rule(alert_rule)
                except Exception:
                    sentry_sdk.capture_exception()
                    raise BadRequest(message="Error when updating alert rule")
```
Does this need to be wrapped in a check for the flag?

Reply:
We enter the dual write method regardless of whether an org is flagged into dual write—in case we need to roll back for some reason, we don't want rules that have already been dual written to go out of sync. If a dual written rule does not exist, then we quit the dual update helper early.

Reply:
🤔 wouldn't this always try to create both sets of models though? don't we need the feature flag to say "dual write or disable dual write"?